src/test/java/de/tum/in/www1/artemis/exercise/quiz/QuizExerciseUtilService.java

```diff
@@ -259,6 +311,28 @@ public QuizExercise createAndSaveExamQuiz(ZonedDateTime startDate, ZonedDateTime
         QuizExercise quizExercise = QuizExerciseFactory.generateQuizExerciseForExam(exerciseGroup);
         QuizExerciseFactory.addQuestionsToQuizExercise(quizExercise);
 
+        for (var quizQuestion : quizExercise.getQuizQuestions()) {
+            if (quizQuestion.getQuizQuestionStatistic() == null) {
+                quizQuestion.initializeStatistic();
+            }
+
+            if (quizQuestion instanceof MultipleChoiceQuestion multipleChoiceQuestion) {
+                quizExerciseService.fixReferenceMultipleChoice(multipleChoiceQuestion);
+                QuizIdAssigner.assignIds(multipleChoiceQuestion.getAnswerOptions());
+                QuizIdAssigner.assignIds(((MultipleChoiceQuestionStatistic) multipleChoiceQuestion.getQuizQuestionStatistic()).getAnswerCounters());
+            }
+            else if (quizQuestion instanceof DragAndDropQuestion dragAndDropQuestion) {
+                quizExerciseService.fixReferenceDragAndDrop(dragAndDropQuestion);
+                quizExerciseService.restoreCorrectMappingsFromIndicesDragAndDrop(dragAndDropQuestion);
+                QuizIdAssigner.assignIds(((DragAndDropQuestionStatistic) dragAndDropQuestion.getQuizQuestionStatistic()).getDropLocationCounters());
+            }
+            else if (quizQuestion instanceof ShortAnswerQuestion shortAnswerQuestion) {
+                quizExerciseService.fixReferenceShortAnswer(shortAnswerQuestion);
+                quizExerciseService.restoreCorrectMappingsFromIndicesShortAnswer(shortAnswerQuestion);
+                QuizIdAssigner.assignIds(((ShortAnswerQuestionStatistic) shortAnswerQuestion.getQuizQuestionStatistic()).getShortAnswerSpotCounters());
+            }
+        }
```

Comment:
You can refactor this method to reuse it, there seems to be a lot of code duplication